Utilities/StateMachine.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Sandbox.ModAPI.Ingame;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Utilities {
+	public class StateMachine {
+		private IEnumerator<bool> _machine;

```

Comment:
Wouldn't these actually be states of the machine, rather than state machines themselves?

Reply:
I still think this is a confusing nomenclature, but if you're adamant about it I'll go ahead and approve.

Reply:
... maybe?

It's not really a proper state machine in the traditional sense, so much as a list of actions to take. I'm definitely open to better names.

But, calling this `_states` suggests, to me, that there's some way of moving from one to another other than by simply reading down the list.

Reply:
State machines are often implemented as a stack, so it'd not entirely farfetched.
You could also call it an Action.

Reply:
I like StepMachine, FWIW.